File: C/01_Recap/02_Broadcast/bcast.c
#include <stdio.h>
#include <mpi.h>

#define TAG 100

int main(int argc, char ** argv)
{

  int rank, nproc, recv;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nproc);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  if (rank == 0) {
    printf("Please enter an integer number:");
    fflush(stdout);
    scanf("%d", &recv);
  }

  MPI_Bcast(&recv);

  printf("On rank %3d MPI_Bcast gives value of %3d\n", rank, recv);

  MPI_Finalize();

}

File: Fortran/01_Recap/02_Broadcast/bcast.f90
PROGRAM reduce

  USE mpi
  IMPLICIT NONE

  INTEGER, PARAMETER :: tag = 100

  INTEGER :: rank, recv
  INTEGER :: nproc, ierr

  CALL MPI_Init(ierr)

  CALL MPI_Comm_size(MPI_COMM_WORLD, nproc, ierr)
  CALL MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)

  IF (rank == 0) THEN
    WRITE(*,'(A)', ADVANCE='NO') 'Please enter an integer number :'
    READ(*,'(2I20)') recv
  END IF

  CALL MPI_Bcast(recv, , , , , )
  PRINT *, 'On rank ', rank, ' MPI_Bcast gives value of ', recv

  CALL MPI_Finalize(ierr)

END PROGRAM reduce
